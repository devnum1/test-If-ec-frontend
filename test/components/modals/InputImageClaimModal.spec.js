import { shallowMount } from '@vue/test-utils';
import InputImageClaimModal from '@/components/modals/InputImageClaimModal.vue';

describe('InputImageClaimModal', () => {
  let wrapper;

  const factory = (computed = {}) => {
    return shallowMount(InputImageClaimModal);
  };

  beforeEach(() => {
    wrapper = factory();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  test('match Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Modal to display input exact text', () => {
    const button = wrapper.find('.input-text');

    expect(button.exists()).toBe(true);

    expect(button.text()).toBe('ライブラリから選択');
  });

  test('Click close icon to call CloseModal() method', () => {
    const closeBtn = wrapper.find('.icon-delete');
    wrapper.vm.closeModal = jest.fn();

    closeBtn.trigger('click');
    expect(wrapper.vm.closeModal).toHaveBeenCalled();
  });

  test('check closeModal() emits and sets value', () => {
    const closeBtn = wrapper.find('.icon-delete');

    closeBtn.trigger('click');

    expect(wrapper.emitted().input.length).toBe(1);
  });
});
