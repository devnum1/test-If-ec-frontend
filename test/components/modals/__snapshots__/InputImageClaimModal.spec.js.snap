// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputImageClaimModal match Snapshot 1`] = `
"<div class=\\"card-payment-instruc-modal\\">
  <div class=\\"modal__overlay\\">
    <div class=\\"modal__container\\">
      <div class=\\"modal__close-button\\">
        <div class=\\"icon-delete\\"></div>
      </div>
      <div class=\\"modal__body\\"><label class=\\"claim-form-image w-100\\">
          <div>
            <div class=\\"claim-form-image-input\\">
              <div class=\\"input-icon\\"><i class=\\"icon-fonts icon-fonts--image\\"></i></div>
              <div class=\\"input-text\\">
                ライブラリから選択
              </div>
            </div>
          </div> <input type=\\"file\\" class=\\"d-none\\">
        </label>
        <div class=\\"claim-form-instruc\\">
          <div class=\\"notice\\">
            <div class=\\"notice-icon\\">
              <div>
                !
              </div>
            </div>
            <div class=\\"notice-text\\">
              写真添付時のご注意
            </div>
          </div>
          <div class=\\"claim-form-instruc--right\\">
            <div class=\\"claim-form-instruc__item\\">
              <div class=\\"image image--right\\"><img src=\\"~/assets/images/form-right.png\\"> <i class=\\"icon-fonts icon-fonts--circle\\"></i></div>
              <div class=\\"instruc-text\\">
                全部が綺麗に写っている
              </div>
            </div>
          </div>
          <div class=\\"claim-form-instruc--wrong\\">
            <div class=\\"claim-form-instruc__item\\">
              <div class=\\"image image--wrong\\"><img src=\\"~/assets/images/form-wrong-1.png\\"> <i class=\\"icon-fonts icon-fonts--close\\"></i></div>
              <div class=\\"instruc-text\\">
                見えない部分がある
              </div>
            </div>
            <div class=\\"claim-form-instruc__item\\">
              <div class=\\"image image--wrong\\"><img src=\\"~/assets/images/form-wrong-2.png\\"> <i class=\\"icon-fonts icon-fonts--close\\"></i></div>
              <div class=\\"instruc-text\\">
                全部写ってない
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
